themes/githelpers.theme.bash

```diff
@@ -101,29 +117,51 @@ function _git-status-counts {
 }
 
 function _git-remote-info {
-  [[ "$(_git-upstream)" == "" ]] && return || true
-
-  [[ "$(_git-branch)" == "$(_git-upstream-branch)" ]] && local same_branch_name=true || true
-  local same_branch_name=
-  [[ "$(_git-branch)" == "$(_git-upstream-branch)" ]] && same_branch_name=true
-  if ([[ "${SCM_GIT_SHOW_REMOTE_INFO}" = "auto" ]] && [[ "$(_git-num-remotes)" -ge 2 ]]) ||
-      [[ "${SCM_GIT_SHOW_REMOTE_INFO}" = "true" ]]; then
-    if [[ "${same_branch_name}" != "true" ]]; then
-      remote_info="\$(_git-upstream)"
-    else
-      remote_info="$(_git-upstream-remote)"
+
+  if [[ "${SCM_GIT_GITSTATUSD_RAN}" == "true" ]]; then # prompt handling only, reimplement because patching the routine below gets ugly
+    [[ "${VCS_STATUS_REMOTE_NAME}" == "" ]] && return || true
+    [[ "${VCS_STATUS_LOCAL_BRANCH}" == "${VCS_STATUS_REMOTE_BRANCH}" ]] && local same_branch_name=true || true
+    local same_branch_name=
+    [[ "${VCS_STATUS_LOCAL_BRANCH}" == "${VCS_STATUS_REMOTE_BRANCH}" ]] && same_branch_name=true
+    if [[ "${SCM_GIT_SHOW_REMOTE_INFO}" = "true" ]] || [[ "${SCM_GIT_SHOW_REMOTE_INFO}" = "auto" ]]; then # no multiple remote support in gitstatusd
```

Comment:
You can replace `]] || [[` with just `||`.

Reply:
done